sam.c

```diff
@@ -522,7 +522,10 @@ static int bam_tag2cigar(bam1_t *b, int recal_bin, int give_warning) // return 0
     cigar0 = bam_get_cigar(b);
     if (bam_cigar_op(cigar0[0]) != BAM_CSOFT_CLIP || bam_cigar_oplen(cigar0[0]) != c->l_qseq) return 0;
     fake_bytes = c->n_cigar * 4;
-    if ((CG = bam_aux_get(b, "CG")) == 0) return 0; // no CG tag
+    int saved_errno = errno;
+    CG = bam_aux_get(b, "CG");
+    errno = saved_errno;
```

Comment:
This should really test for and propagate the `errno == EINVAL` case when `GC == NULL`, instead of glossing over it.

Reply:
Ok, something like

```
if (errno == ENOENT) errno = saved_errno;
```

I'll amend when I'm finished with my current other PR, or make the trivial change during merge if you wish.

Reply:
```
if (!CG) {
    if (errno == EINVAL) return -1;  // Bad aux data
    errno = saved_errno;
    return 0; // no CG tag
}
```